io.openems.backend.timedata.aggregatedinflux/readme.adoc

```diff
@@ -0,0 +1,122 @@
+bundle can mitigate this bottleneck. This happens by writing all data as usual via xref:../io.openems.backend.timedata.influx.adoc[Timedata Influx] into
+an Influx database (`DB 1`). Furthermore, all "aggregated data" (see xref:timedata.adoc[edge datatypes]) are written via the `Aggregated InfluxDB` bundle to
+another database (`DB 2`) located on another server.
+Thus `DB 1` holds all data in high resolution, while `DB 2` only holds very few data (5-minute averages for power values
+ or only daily values for energy). Furthermore, a retention policy of e.g. 90 days is applied to `DB 2`.
+Thus `DB 1` provides relatively slow access to a huge amount of data
+and `DB 2` provides fast and responsive access to a relatively small database.
+
+Within OpenEMS backend access to both databases is managed by the
+https://github.com/OpenEMS/openems/blob/develop/io.openems.backend.core/src/io/openems/backend/core/timedatamanager/TimedataManagerImpl.java[TimedataManager].
+The TimedataManager writes edge relevant data to **all** Timedata providers,
+whereas it reads data only from the **first** Timedata provider,
+ which can deliver.
+Assuming correct configuration, TimedataManager first returns historical data
+from `DB 2` (fast and responsive).
+And only in a very few cases it gets the data from `DB 1`
+(probably slower).
+
+
+
+
+[CAUTION]
+====
+The `Timedata.AggregatedInfluxDB` includes a class `AllowedChannels.java`.
+This implementation includes a *hardcoded* list of channels:
+
+* `ALLOWED_AVERAGE_CHANNELS` are used to calculate the 5min average values (e.g. average power values).
+* `ALLOWED_CUMULATED_CHANNELS` are used to calculate daily values (e.g. energy values).
+
+This list must be adopted to a concrete usecase. It strongly depends on
+
+* your strategy to select component-IDs.
+* the components you are using within OpenEMS.
+
+If you detect some widgets within your OpenEMS-UI which hava empty values,
+it may have to do with an incorrect hardcoded list.
+
+====
+
+== Configuration Setup
+
+If you expect your backend to handle thousands of connected edges,
+the following configuration may provide a good start setup:
+
+*Create database and set retention policy:*
+
+Before starting the OpenEMS backend and after intially setting up the influx servers,
+you need to create the databases `influx0` and `aggregated0`
+and some retention policies:
+
+[source,shell]
+----
+
+##### Influx Server 1 #####
+curl -i -XPOST http://127.0.0.1:8082/query --data-urlencode "q=CREATE DATABASE influx0"
+
+##### Influx Server 2 #####
+curl -i -XPOST http://127.0.0.1:8081/query --data-urlencode "q=CREATE DATABASE aggregated0"
+
+curl -i -XPOST http://127.0.0.1:8081/query --data-urlencode "q=CREATE RETENTION POLICY rp_max ON aggregated0 DURATION 90d REPLICATION 1"
+curl -i -XPOST http://127.0.0.1:8081/query --data-urlencode "q=CREATE RETENTION POLICY rp_avg ON aggregated0 DURATION 90d REPLICATION 1"
```

Comment:
Maybe we should link to [this](https://docs.influxdata.com/influxdb/v1.8/query_language/manage-database/#create-retention-policies-with-create-retention-policy) documentation. And the more important configuration is the the [shard duraction](https://docs.influxdata.com/influxdb/v1.8/query_language/manage-database/#shard-duration)

If someone has the same settings for both retention policies they dont have to create the same one twice.

We are using two different retention policies:
For avg there should be data every 5-Minutes and the data in this measurement gets used for queries over one day and one week. Therefore we set the shard duration to 24h to only read two shards for a day query.

For max we used a shard duration fo 31 days because there is much less data in this measurement (only one per day) and to optimize queries for months only 2 shards need to be readed by the db.

Reply:
Thanks. This is a valid comment, but in order to release these docs more quickly I decided to leave it out for now. I am happy to accept future documentation PRs from you. ;-)